sympy/discrete/convolutions.py

```diff
@@ -89,7 +90,30 @@ def convolution(a, b, cycle=0, dps=None, prime=None, dyadic=None, subset=None):
     elif subset:
         ls = convolution_subset(a, b)
     else:
-        ls = convolution_fft(a, b, dps=dps)
+        def loop(a):
+            dens = []
+            for i in a:
+                if isinstance(i, Rational) and i.q - 1:
+                    dens.append(i.q)
+                elif not isinstance(i, int):
+                    return
+            if dens:
+                lcm = ilcm(*dens) if len(dens) > 1 else dens[0]
+                return [int(i*lcm) for i in a], lcm
```

Comment:
```suggestion
                return [i.p*lcm//i.q for i in a], lcm
```

Reply:
Will this work if `a` holds any python `int`s?